### PR TITLE
feat(sql): replaced `SNAPSHOT` SQL with `CHECKPOINT` to reduce confusion

### PR DESCRIPTION
**Pull request by bluestreak01**
Legacy SQLs:

```sql
snaphot prepare;
snapshot complete;
```

New SQL:
```sql
checkpoint create
checkpoint release
```

Considering our backup system works in tandem with cloud or file system's SNAPSHOT functionality it became confusing to document which snapshot we're talking about when.

We are moving our terminology to use "checkpoint".

### User impact

This is a **NON BREAKING** chance. Legacy SQL will still work, although it won't be documented. The snapshots created on older QuestDB versions will still be restored. by new version.

### TODO

- [x] add backward compatibility tests

### Misc

CI tripped over https://github.com/questdb/questdb/issues/4823

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 63 / 65 (96.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/Bootstrap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/WorkerPoolUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgentImpl.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 90 / 100 (90.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgentImpl.java|38|45|84.44%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/Bootstrap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/WorkerPoolUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 101 / 121 (83.47%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|15|25|60.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgentImpl.java|41|48|85.42%|
|:large_blue_circle:|io/questdb/Bootstrap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/WorkerPoolUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 111 / 129 (86.05%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|15|19|78.95%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgent.java|41|48|85.42%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|24|28|85.71%|
|:large_blue_circle:|io/questdb/Bootstrap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 123 / 142 (86.62%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|15|19|78.95%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|8|10|80.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|24|28|85.71%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgent.java|47|54|87.04%|
|:large_blue_circle:|io/questdb/Bootstrap.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 123 / 141 (87.23%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|15|19|78.95%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|8|10|80.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|24|28|85.71%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgent.java|47|54|87.04%|
|:large_blue_circle:|io/questdb/Bootstrap.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 122 / 141 (86.52%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|15|19|78.95%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|8|10|80.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|24|28|85.71%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgent.java|47|54|87.04%|
|:large_blue_circle:|io/questdb/Bootstrap.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

File: core/src/test/java/io/questdb/test/PropServerConfigurationTest.java [hunk 1]
```diff
@@ -287,10 +287,10 @@ public void testAllDefaults() throws Exception {
 
         TestUtils.assertEquals(new File(root, "db").getAbsolutePath(), configuration.getCairoConfiguration().getRoot());
         TestUtils.assertEquals(new File(root, "conf").getAbsolutePath(), configuration.getCairoConfiguration().getConfRoot());
-        TestUtils.assertEquals(new File(root, "snapshot").getAbsolutePath(), configuration.getCairoConfiguration().getSnapshotRoot());
+        TestUtils.assertEquals(new File(root, TableUtils.CHECKPOINT_DIRECTORY).getAbsolutePath(), configuration.getCairoConfiguration().getCheckpointRoot());
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/PropServerConfigurationTest.java:290]: nit: would be nice to keep an assertion for the legacy dir.

File: core/src/test/java/io/questdb/test/griffin/CheckpointTest.java [hunk 2]
```diff
@@ -49,11 +49,12 @@
 
 import static io.questdb.PropertyKey.CAIRO_SNAPSHOT_RECOVERY_ENABLED;
 
-public class SnapshotTest extends AbstractCairoTest {
+public class CheckpointTest extends AbstractCairoTest {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/griffin/CheckpointTest.java:52]: There are no tests here that would call `snapshot complete;`, so backward compatibility may be broken in the future without us noticing.

File: core/src/main/java/io/questdb/cairo/CairoConfiguration.java [hunk 3]
```diff
@@ -354,7 +354,9 @@ default RostiAllocFacade getRostiAllocFacade() {
     CharSequence getSnapshotInstanceId();
 
     @NotNull
-    CharSequence getSnapshotRoot(); // same as root/../snapshot
+    CharSequence getCheckpointRoot(); // same as root/../.checkpoint
+
+    CharSequence getLegacyCheckpointRoot(); // same as root/../snapshot
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/CairoConfiguration.java:359]: fixed

File: core/src/main/java/io/questdb/PropServerConfiguration.java [hunk 4]
```diff
@@ -640,20 +640,22 @@ public PropServerConfiguration(
...
+            this.legacyCheckpointRoot = new File(root, TableUtils.LEGACY_CHECKPOINT_DIRECTORY).getAbsolutePath();
             tmpRoot = new File(root, TMP_DIRECTORY).getAbsolutePath();
         }
 
         this.cairoAttachPartitionSuffix = getString(properties, env, PropertyKey.CAIRO_ATTACH_PARTITION_SUFFIX, TableUtils.ATTACHABLE_DIR_MARKER);
         this.cairoAttachPartitionCopy = getBoolean(properties, env, PropertyKey.CAIRO_ATTACH_PARTITION_COPY, false);
 
         this.snapshotInstanceId = getString(properties, env, PropertyKey.CAIRO_SNAPSHOT_INSTANCE_ID, "");
-        this.snapshotRecoveryEnabled = getBoolean(properties, env, PropertyKey.CAIRO_SNAPSHOT_RECOVERY_ENABLED, true);
+        this.checkpointRecoveryEnabled = getBoolean(properties, env, PropertyKey.CAIRO_SNAPSHOT_RECOVERY_ENABLED, true);
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/PropServerConfiguration.java:658]: done

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 5]
```diff
@@ -202,7 +204,7 @@ public static int calculateTxnLagChecksum(long txn, long seqTxn, int lagRowCount
         checkSum = checkSum * 31 + seqTxn;
         checkSum = checkSum * 31 + lagRowCount;
         checkSum = checkSum * 31 + lagTxnCount;
-        return (int) (checkSum ^ (checkSum >>> 32));
+        return Long.hashCode(checkSum);
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:207]: reverted

File: core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java [hunk 6]
```diff
@@ -1628,6 +1643,21 @@ private void compileInner(@Transient @NotNull SqlExecutionContext executionConte
         compiledQuery.withContext(executionContext);
     }
 
+    private void compileLegacySnapshot(SqlExecutionContext executionContext) throws SqlException {
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java:1646]: sure

File: core/src/main/java/io/questdb/Bootstrap.java [hunk 7]
```diff
@@ -537,7 +537,8 @@ private void reportValidateConfig() {
         try (Path path = new Path()) {
             verifyFileSystem(path, cairoConfig.getRoot(), "db", true);
             verifyFileSystem(path, cairoConfig.getBackupRoot(), "backup", true);
-            verifyFileSystem(path, cairoConfig.getSnapshotRoot(), "snapshot", true);
+            verifyFileSystem(path, cairoConfig.getCheckpointRoot(), TableUtils.CHECKPOINT_DIRECTORY, true);
+            verifyFileSystem(path, cairoConfig.getCheckpointRoot(), TableUtils.LEGACY_CHECKPOINT_DIRECTORY, true);
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/Bootstrap.java]: done https://github.com/questdb/questdb/pull/4821/commits/f3ee3c9b634661a99ae231e16470ca77c51a3d65

File: core/src/test/java/io/questdb/test/PropServerConfigurationTest.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/test/PropServerConfigurationTest.java:291]: done https://github.com/questdb/questdb/pull/4821/commits/f3ee3c9b634661a99ae231e16470ca77c51a3d65

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 8]
```diff
@@ -100,9 +102,9 @@ public final class TableUtils {
     public static final String META_SWAP_FILE_NAME = "_meta.swp";
     public static final int MIN_INDEX_VALUE_BLOCK_SIZE = Numbers.ceilPow2(4);
     public static final int NULL_LEN = -1;
-    public static final String RESTORE_SNAPSHOT_TRIGGER_FILE_NAME = "_restore";
-    public static final String SNAPSHOT_META_FILE_NAME = "_snapshot";
-    public static final String SNAPSHOT_META_FILE_NAME_TXT = "_snapshot.txt";
+    public static final String RESTORE_FROM_CHECKPOINT_TRIGGER_FILE_NAME = "_restore";
+    public static final String CHECKPOINT_META_FILE_NAME = "_snapshot";
+    public static final String CHECKPOINT_META_FILE_NAME_TXT = "_snapshot.txt";
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:107]: we do not actually write _snapshot.txt anywhere in the code, this is a manual workaround hack to trigger snapshot recovery when snapshot IDs do not match. I will leave this file as is. It is not used by the new trigger

[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableUtils.java]: changed in https://github.com/questdb/questdb/pull/4821/commits/ffc3d3369dbec16d1256afc54ec50fdcc88cf310

File: core/src/test/java/io/questdb/test/griffin/wal/CheckpointFuzzTest.java [hunk 9]
```diff
@@ -121,21 +127,58 @@ public void testSnapshotFrequentTableDrop() throws Exception {
...
     }
 
+    private void checkpointCreate() throws SqlException {
+        LOG.info().$("creating checkpoint").$();
+
+        ddl("checkpoint create");
+        CairoConfiguration conf = engine.getConfiguration();
+
+        FilesFacade ff = conf.getFilesFacade();
+        Path snapshotPath = Path.getThreadLocal(conf.getRoot()).put("_snapshot");
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/test/java/io/questdb/test/griffin/wal/CheckpointFuzzTest.java]: changed in https://github.com/questdb/questdb/pull/4821/commits/d2f286e0b1a29c2bd7dd8f54c085766ace8a89ab

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 10]
```diff
@@ -100,9 +102,10 @@ public final class TableUtils {
     public static final String META_SWAP_FILE_NAME = "_meta.swp";
     public static final int MIN_INDEX_VALUE_BLOCK_SIZE = Numbers.ceilPow2(4);
     public static final int NULL_LEN = -1;
-    public static final String RESTORE_SNAPSHOT_TRIGGER_FILE_NAME = "_restore";
-    public static final String SNAPSHOT_META_FILE_NAME = "_snapshot";
-    public static final String SNAPSHOT_META_FILE_NAME_TXT = "_snapshot.txt";
+    public static final String RESTORE_FROM_CHECKPOINT_TRIGGER_FILE_NAME = "_restore";
+    public static final String CHECKPOINT_LEGACY_META_FILE_NAME = "_snapshot";
+    public static final String CHECKPOINT_LEGACY_META_FILE_NAME_TXT = "_snapshot.txt";
+    public static final String CHECKPOINT_META_FILE_NAME = "_checkpoint_mata.d";
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:108]: Typo here:
```suggestion
    public static final String CHECKPOINT_META_FILE_NAME = "_checkpoint_meta.d";
```

File: core/src/main/java/io/questdb/PropertyKey.java [hunk 11]
```diff
@@ -33,8 +33,9 @@ public enum PropertyKey implements ConfigPropertyKey {
     BINARYDATA_ENCODING_MAXLENGTH("binarydata.encoding.maxlength"),
     CAIRO_ROOT("cairo.root"),
     CAIRO_VOLUMES("cairo.volumes"),
-    CAIRO_SNAPSHOT_INSTANCE_ID("cairo.snapshot.instance.id"),
+    CAIRO_LEGACY_SNAPSHOT_INSTANCE_ID("cairo.snapshot.instance.id"),
     CAIRO_SNAPSHOT_RECOVERY_ENABLED("cairo.snapshot.recovery.enabled"),
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/PropertyKey.java:37]: nit: might be a good idea to add legacy to the enum name: `CAIRO_LEGACY_SNAPSHOT_RECOVERY_ENABLED`

File: core/src/main/java/io/questdb/cairo/TableUtils.java (hunk 10 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:108]: thanks Andrei :)

File: core/src/main/java/io/questdb/PropertyKey.java (hunk 11 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/PropertyKey.java]: done https://github.com/questdb/questdb/pull/4821/commits/54be271306f73d2a04f6caf59638838b561fbaa1

File: core/src/main/java/io/questdb/cairo/TableUtils.java (hunk 10 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cairo/TableUtils.java]: fixed in https://github.com/questdb/questdb/pull/4821/commits/54be271306f73d2a04f6caf59638838b561fbaa1

File: core/src/test/java/io/questdb/test/griffin/CheckpointTest.java (hunk 2 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/test/griffin/CheckpointTest.java:53]: added exercising of legacy commands to checkpoint fuzz test here https://github.com/questdb/questdb/pull/4821/commits/54be271306f73d2a04f6caf59638838b561fbaa1

**Comment by bluestreak01** [comment]
i think I addressed all of the comments @puzpuzpuz

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 132 / 151 (87.42%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|2|3|66.67%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|15|19|78.95%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/griffin/SqlCompilerImpl.java|24|28|85.71%|
|:large_blue_circle:|io/questdb/cairo/DatabaseCheckpointAgent.java|53|60|88.33%|
|:large_blue_circle:|io/questdb/Bootstrap.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3PartitionPurgeJob.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/CompiledQueryImpl.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/ColumnPurgeJob.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/ShowTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/catalogue/InformationSchemaTablesFunctionFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|2|2|100.00%|

**Review by puzpuzpuz — APPROVED** [review]

**Pull request merged**